### PR TITLE
nodejs-16_x: 16.5.0 -> 16.6.1

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change
https://github.com/nodejs/node/releases/tag/v16.6.0
https://github.com/nodejs/node/releases/tag/v16.6.1

cc: @happysalada 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
You're amazing for getting to this.
I'm building on local to confirm all is working.
Will update soon.

File: pkgs/development/web/nodejs/v16.nix [hunk 1]
```diff
@@ -8,14 +8,7 @@ let
...
-      # Fix CVE-2021-22930 https://github.com/nodejs/node/pull/39423.
-      # It should be fixed by Node.js 16.6.0, but currently it fails to build on Darwin
-      (fetchpatch {
-        url = "https://github.com/nodejs/node/commit/9d950a0956bf2c3dd87bacb56807f37e16a91db4.patch";
-        sha256 = "1narhk5dqdkbndh9hg0dn5ghhgrd6gsamjqszpivmp33nl5hgsx3";
-      })
-    ];
+    version = "16.6.1";
+    sha256 = "0mz5wfhf2k1qf3d57h4r8b30izhyg93g5m9c8rljlzy6ih2ymcbr";
+    patches = lib.optionals stdenv.isDarwin [ ./disable-darwin-v8-system-instrumentation.patch ];
```
Comment (SuperSandro2000) [inline pkgs/development/web/nodejs/v16.nix:13]:
```suggestion
    patches = [ ./disable-darwin-v8-system-instrumentation.patch ];
```
This shouldn't break anything I think.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133232` at 59a3000e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>lumo</li>
    <li>nodejs-slim-16_x (nodejs-slim_latest)</li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>nodejs-16_x (nodejs_latest)</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/web/nodejs/v16.nix:13:47:
```
   |
13 |     patches = lib.optionals stdenv.isDarwin [ ./disable-darwin-v8-system-instrumentation.patch ];
   |                                               ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md">name-and-version</a>

Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/development/web/nodejs/nodejs.nix:48:5:
```
   |
48 |     name = "${baseName}-${version}";
   |     ^
```

Near pkgs/development/web/nodejs/nodejs.nix:46:12:
```
   |
46 |     inherit version;
   |            ^
```
</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 133232` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>lumo</li>
    <li>nodejs-16_x (nodejs_latest)</li>
    <li>nodejs-slim-16_x (nodejs-slim_latest)</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Tests pass, and it built successfully.
Thank you again!

**Comment by pieromqwerty** [comment]
Wow you're amazing for this! I was trying to do it last night but was unsuccessful

**Pull request merged**